### PR TITLE
Added naming convention document

### PR DESCRIPTION
**Pull request by atsikham**
Task #2023

### CODE REVIEW

**Review by mkyc — COMMENTED** [review]
where would vm_groups part go?

**Comment by atsikham** [comment]
> where would vm_groups part go?

That's added after the prefix for both BI modules, added an information about that.

File: docs/design-docs/infrastructure/naming-convention.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Prefix
+
+Epiphany modules are developed in the way that allows user specify a prefix for created resources. This approach gives
+such benefits as ordered sorting and identifying who is the owner of the resource. Prefix can include following parts
+with a dash `-` as a delimiter.
+
+| Type | Required | Description | Examples |
+| ---- | -------- | ----------- | -------- |
+| Owner | yes | The name of the person or team which resource belongs to | epiphany |
+| Application or service name | yes | Name of the application, workload, or service that the resource is a part of | kafka, ignite, opendistro |
```
Comment (mkyc) [inline docs/design-docs/infrastructure/naming-convention.md]: One more question regarding required=yes here. I'm not sure, but that doesn't look like required to me. Is resource group part of some application for example? I believe owner is one required field for sure. 

At second look at this document it also looks that vm_group is just one of elements of prefix.

Comment (atsikham) [reply docs/design-docs/infrastructure/naming-convention.md]: Please check now

**Review by mkyc — APPROVED** [review]
I think it's good policy now. It's clear, short, general and extendable.

**Review by seriva — APPROVED** [review]
For Azure this is fine, but how about AWS?  IMO more problematic as we dont have anything like a proper resource type of thing to group everything under. Note that this can be in a seperate issue for me, hense my approval.

edit: also note that not all resources can follow this conventions like storage account name: https://github.com/epiphany-platform/epiphany/blob/90b55f8d9dc44586217fc5035b855e9b89bfa02e/core/src/epicli/cli/helpers/naming_helpers.py#L34

**Comment by atsikham** [comment]
> For Azure this is fine, but how about AWS? IMO more problematic as we dont have anything like a proper resource type of thing to group everything under. Note that this can be in a seperate issue for me, hense my approval.
> 
> edit: also note that not all resources can follow this conventions like storage account name:

As I remember this issue originally came from modules, when we started naming in different ways, without care about delimiters, index and resource type positions. I checked AWS BI module and don't see resources there that can't follow these rules.

In my opinion we can try following these recommendations and update the document if something wasn't taken into account.

**Pull request merged**